pandas/core/arrays/categorical.py

```diff
@@ -520,6 +520,9 @@ def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
             if dtype == self.dtype:
                 return self
             return self._set_dtype(dtype)
+        if is_integer_dtype(dtype) and self.isna().any():
+            msg = "Cannot convert float NaN to integer"
```

Comment:
Kinda pedantic, but we can have other NA values here.

```python
In [18]: cat = pd.Categorical([pd.Timestamp('2000'), pd.NaT])
```

In this case, it's not a float that we're refusing to cast. So perhaps `Cannot convert NA to integer`.

Reply:
Agree this seems more correct; do we want to think about consistency with `Series` as @jreback pointed out above?  Incidentally I just noticed that `Series` seems to be misbehaving as well in this special case, so probably worth a separate issue or PR:
```python
[ins] In [5]: pd.Series([pd.Timestamp("2000"), pd.NaT]).astype(int)             
Out[5]: 
0     946684800000000000
1   -9223372036854775808
dtype: int64
```